### PR TITLE
Use site domain instead of blog_id ton wp.com stats link.

### LINKED ISSUE #2222: Stats: Link to WordPress.com needs updating
Stats on WP.com now default to domain name, instead of ID. Once 3307 is resolved, we should update to match current convention.

https://github.com/Automattic/jetpack/blob/07bf09d986c3e8d35f583965c60aaaebef1722d5/modules/stats.php#L372

### PR DESCRIPTION
**Pull request by osiux**
Fixes #2222 

I used something similar to where we [get $domain on another module](https://github.com/Automattic/jetpack/blob/e92880156b40ff407d887a694f2eae59087c356e/modules/likes.php#L865-L874).

### CODE REVIEW

File: modules/stats.php [hunk 1]
```diff
@@ -370,7 +370,16 @@ function stats_reports_page() {
 	if ( isset( $_GET['dashboard'] ) )
 		return stats_dashboard_widget_content();
 
-	$blog_id = stats_get_option( 'blog_id' );
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$blog_id = get_current_blog_id();
+		$bloginfo = get_blog_details( (int) $blog_id );
+		$domain = $bloginfo->domain;
+	} else {
+		$blog_id = Jetpack_Options::get_option( 'id' );
```
Comment (dereksmart) [inline modules/stats.php]: Doesn't look like you're using the `$blog_id` in this part of the conditional

File: modules/stats.php [hunk 2]
```diff
@@ -370,7 +370,16 @@ function stats_reports_page() {
...
+		$url = home_url();
+		$url_parts = parse_url( $url );
+		$domain = $url_parts['host'];
```
Comment (dereksmart) [inline modules/stats.php]: We have a handly method `Jetpack::build_raw_urls( $url )` that we can use here.  

`Jetpack::build_raw_urls( get_home_url() )` should do the trick.

**Comment by osiux** [comment]
@dereksmart nice, didn't know that method. Something like this? I leaved $blog_id because it's used down in the file.

**Comment by dereksmart** [comment]
Note that this only changes the link that appears when the stats page is loading 
screen shot 2015-11-03 at 12 13 45 pm

We need to update the iframed page from wpcom to change the "show me" button link, but this PR is good to go.

**Pull request merged**